docs/toml.md

```diff
@@ -133,6 +128,235 @@ Supported filters:
+Using the default CLF option is simple, e.g.
+
+```
+[accessLog]
+  file = "access.log"
+```
+
+The file can be gzipped as it is written if you require; just add a
+`.gz` extension:
+
+```
+[accessLog]
+  file = "access.log.gz"
+```
+
+The timestamps written to the log file are in the CLF pattern. This can be overridden
+using the `timeFormat` setting, e.g.
+
+```
+[accessLog]
+  file       = "access.log"
+  timeFormat = "02/Jan/2006:15:04:05 -0700"
+```
+
+explicitly sets the CLF pattern, which is the default.
+
+### Buffered writing
+
+Access logs are by default written using a one-line buffer. You can increase
+the buffering, e.g.
+```
+[accessLog]
+  file       = "access.log"
+  bufferSize = 4 KiB
+```
+which improves the write performance but there is a slight delay before some
+items are written. Note however that the buffer size is ignored if you have
+a gzipped output file (gzipping also implies buffering of its own).
+
+### Asynchronous writing
+
+Access logs are by default written by the goroutine (similar to a thread) that
+is handling the HTTP request round-trip. You can specify that a separate goroutine
+is to be used, e.g.
+```
+[accessLog]
+  file          = "access.log"
+  async         = true
+  channelBuffer = 100
+```
+The `async` flag enables this feature and `channelBuffer` specifies how many
+log records can be stacked up waiting to be written to file.
+
+Asynchronous writing can be combined with buffered or gzipped writing if required.
+
+Using asynchronous writing will reduce the latency of the HTTP request round-trip
+by taking out the file writing. However, overall there will be a greater CPU load.
+
+The channel buffer has a fixed upper limit. If it ever becomes full, no log records
+will be lost but there will be back-pressure. So, under heavy load in which it becomes
+full, the back pressure will cause the HTTP request round-trip latency to be impaired.
+
+Therefore `channelBuffer` should be just large enough to minimise unwanted
+back-pressure, but should otherwise be as small as possible so that the overhead of
+managing its memory is not too great.
+
+Note that this effect can show up if you do performance testing: under saturation
+loading, asynchronous log writing will perform worse than direct log writing because
+of back pressure. Under unsaturated loading, asynchronous log writing could perform
+better.
+
+If you are unsure, don't use asynchronous writing; buffering (or gzipping) gives good
+speed anyway.
+
+### Access logs in JSON
+
+If you need logs in JSON, set `format = "json"` and, optionally, define the four blocks
+of field mappings, which are
+
+ * `coreFields` : data gleaned from the HTTP request.
+ * `requestHeaders` : the set of headers received in the request.
+ * `originResponseHeaders` : the set of headers received from the origin (a.k.a. upstream) server
+ * `downstreamResponseHeaders` : the set of headers passed back to the client
```

Comment:
There are various reasons for them to diverge, according to what the middleware layers do. The obvious example is the compressor, which changes the content length. Other middlewares might explicitly set response headers, e.g. for cache control.